controllers/storagecluster/external_resources.go

```diff
@@ -248,36 +287,13 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 		objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
 		switch d.Kind {
 		case "CephCluster":
-			monitoringIP := d.Data["MonitoringEndpoint"]
-			if monitoringIP == "" {
-				err := fmt.Errorf(
-					"Monitoring Endpoint not present in the external cluster secret %s",
-					externalClusterDetailsSecret)
-				r.Log.Error(err, "Failed to get Monitoring IP.")
-				return err
-			}
-			// replace any comma in the monitoring ip string with space
-			// and then collect individual (non-empty) items' array
-			monIPArr := parseMonitoringIPs(monitoringIP)
-			monitoringPort := d.Data["MonitoringPort"]
-			if monitoringPort != "" {
-				var err error
-				for _, eachMonIP := range monIPArr {
-					err = checkEndpointReachable(net.JoinHostPort(eachMonIP, monitoringPort), 5*time.Second)
-					// if any one of the mon's IP:PORT combination is reachable,
-					// consider the whole set as valid
-					if err == nil {
-						break
-					}
-				}
-				if err != nil {
-					r.Log.Error(err, "Monitoring validation failed")
-					return err
-				}
-				r.monitoringPort = monitoringPort
+			// nothing to be done here,
+			// as all the validation will be done in CephCluster creation
+			if d.Name == "monitoring-endpoint" {
+				continue
```

Comment:
Why not do the validations here?
Is there a benefit in moving it to CephCluster creation block?

Reply:
IIRC this PR fixes a particular bug, so we should avoid making unrelated changes (in case we need to backport and maintain). Refactors should be kept separate from bug fixes.

Reply:
I have moved the functionality to a different PR.

IMO we should not be doing the verification/validation at one function and  use it entirely in a different function (expecting the first function to be called as the second function is not directly calling the first function).

Let me know we really want to put the verification/validation in the previous place.

There is always a question why not use it in TWO places. But our current testing workflow prevents it.

Reply:
Will create a backport manually once it is merged

Reply:
:+1: